experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -202,6 +205,76 @@ export abstract class LazyField<TKind extends FieldKindTypes, TTypes extends All
 		);
 	}
 
+	public reduce<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: UnboxNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => U | Stop<U>,
+		initialValue?: U | undefined,
+	): UnboxNodeUnion<TTypes> | U {
+		return this.reduceImpl(
+			callbackfn,
+			(cursor) => unboxedUnion(this.context, this.schema, cursor),
+			initialValue,
+		);
+	}
+
+	public reduceBoxed<U>(
+		callbackfn: (
+			previousValue: U,
+			currentValue: TypedNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => U | Stop<U>,
+		initialValue?: U | undefined,
+	): TypedNodeUnion<TTypes> | U {
+		return this.reduceImpl(
+			callbackfn,
+			(cursor) => makeTree(this.context, cursor) as TypedNodeUnion<TTypes>,
+			initialValue,
+		);
+	}
```

Comment:
@CraigMacomber - The desire is eliminate boxing as a concept from the public API.  I understand that Nick agreed to hold off on removing pre-existing code in case we find a scenario where we we're unable to think of an alternative, but we should avoid adding/exposing new boxed APIs in the interim.